server/jetstream_test.go

```diff
@@ -19644,3 +19643,76 @@ func TestJetStreamPullConsumerLastPerSubjectRedeliveries(t *testing.T) {
 		m.Ack()
 	}
 }
+
+func TestJetStreamPullConsumersTimeoutHeaders(t *testing.T) {
+	s := RunBasicJetStreamServer()
+	if config := s.JetStreamConfig(); config != nil {
+		defer removeDir(t, config.StoreDir)
+	}
+	defer s.Shutdown()
+
+	// Client for API requests.
+	nc, js := jsClientConnect(t, s)
+	defer nc.Close()
+
+	_, err := js.AddStream(&nats.StreamConfig{
+		Name:     "TEST",
+		Subjects: []string{"foo"},
+	})
+	require_NoError(t, err)
+
+	_, err = js.AddConsumer("TEST", &nats.ConsumerConfig{
+		Durable:       "dlc",
+		AckPolicy:     nats.AckExplicitPolicy,
+		FilterSubject: "foo",
+	})
+	require_NoError(t, err)
+
+	nc.Publish("foo", []byte("foo"))
+	nc.Publish("foo", []byte("bar"))
+	nc.Publish("foo", []byte("baz"))
+	nc.Flush()
+
+	// We will do low level requests by hand for this test as to not depend on any client impl.
+	rsubj := fmt.Sprintf(JSApiRequestNextT, "TEST", "dlc")
+
+	req := &JSApiConsumerGetNextRequest{Batch: 50, Expires: 100 * time.Millisecond, NoWait: false, MaxBytes: 1024}
+	jreq, err := json.Marshal(req)
+	require_NoError(t, err)
+	// Create listener.
+	reply, msgs := nats.NewInbox(), make(chan *nats.Msg, 50)
+	sub, err := nc.ChanSubscribe(reply, msgs)
+	require_NoError(t, err)
+	defer sub.Unsubscribe()
+
+	// Send request.
+	err = nc.PublishRequest(rsubj, reply, jreq)
+	require_NoError(t, err)
+
+	bytesReceived := 0
+	messagesReceived := 0
+	for {
+		select {
+		case m := <-msgs:
+			if len(m.Data) == 0 && m.Header != nil {
+
+				if value := m.Header.Get("Nats-Pending-Messages"); value != fmt.Sprint(50-messagesReceived) {
+					t.Fatalf("Expected %d messages, got %s", 50-messagesReceived, value)
+				}
+				fmt.Println("bytes: ", m.Header)
+				fmt.Println("byets received", bytesReceived)
+				if value := m.Header.Get("Nats-Pending-Bytes"); value != fmt.Sprint(1024-bytesReceived) {
+					t.Fatalf("Expected %d bytes, got %s", 1024-bytesReceived, value)
+				}
+				return
+			} else {
+				messagesReceived += 1
+				fmt.Printf("data: %d, headers %d reply %d\n", len(m.Data), len(m.Header), len(m.Reply))
```

Comment:
Again no print once you are done debugging.

Reply:
Yes, will clean up printlns after being done with tests.